### ISSUE TITLE
Continuing lora training from checkpoint?

### ISSUE THREAD

**Issue by burgerlawful** [oobabooga/text-generation-webui#827]
I accidentally exited my training with keyboard interrupt in conda instead of pressing the interrupt button in the webui so it didn't save the lora, but I still have a few automatically saved training checkpoints, is it possible to continue training from one of them somehow?

**Comment by oobabooga** [comment]
The training code has been completely rewritten in v4.0. Checkpoint resumption is now supported: HF Trainer checkpoint directories are detected automatically and training resumes with full optimizer/scheduler state. See the [release notes](https://github.com/oobabooga/text-generation-webui/releases/tag/v4.0) and [training docs](https://github.com/oobabooga/text-generation-webui/wiki/05-%E2%80%90-Training-Tab).
